### PR TITLE
EOS-24987 : Removal of temporary code

### PR DESCRIPTION
**Pull request by ydb242**
# Problem Statement
- Removal of unused timer code in m0_btree_close()

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -6973,51 +6956,29 @@ static int64_t btree_close_tree_tick(struct m0_sm_op *smop)
...
+		/* Fallthrough to P_ACT*/
 
+	case P_ACT:
 		m0_rwlock_write_lock(&tree->t_lock);
 		if (tree->t_ref > 1) {
 			m0_rwlock_write_unlock(&tree->t_lock);
-			bnode_put(tree->t_root->n_op, tree->t_root);
-			tree_put(tree);
-			return P_DONE;
+			check = 1;
```
Comment (shashank-parulekar) [inline btree/btree.c:6973]: The code should do the way it was before except for the fact that we do not time out any more and instead we keep waiting till the reference count goes to 0.

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: btree/btree.c (hunk 1 above)
Comment (ydb242) [reply btree/btree.c]: Done

File: btree/btree.c [hunk 2]
```diff
@@ -6989,33 +6980,22 @@ static int64_t btree_close_tree_tick(struct m0_sm_op *smop)
 			tree_put(tree);
 			return P_DONE;
 		}
-		tree->t_starttime = m0_time_now();
 		m0_rwlock_write_unlock(&tree->t_lock);
 
 		/** put tree's root node. */
 		bnode_put(tree->t_root->n_op, tree->t_root);
 		/** Fallthrough to P_TIMECHECK */
 	case P_TIMECHECK:
```
Comment (NikhilBirgade) [inline btree/btree.c:6988]: Since we are not checking time anymore, you might have to change the name too.

Comment (ydb242) [reply btree/btree.c]: Changed the name to WAITCHECK.

**Review by NikhilBirgade — APPROVED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**